extensions/poetry/main.ts

```diff
@@ -0,0 +1,108 @@
+import { PhylumApi } from "phylum"
+import { parseDryRun } from "./parse.ts"
+import { red, green, yellow } from "https://deno.land/std@0.150.0/fmt/colors.ts";
+
+class FileBackup {
+  readonly fileName: string
+  readonly fileContent: string | null
+
+  constructor(fileName: string) {
+    this.fileName = fileName
+    this.fileContent = null
+  }
+
+  async backup() {
+    try {
+      this.fileContent = await Deno.readTextFile(this.fileName)
+    } catch (e) { }
+  }
+
+  async restoreOrDelete() {
+    try {
+      if (this.fileContent != null) {
+        await Deno.writeTextFile(this.fileName, this.fileContent)
+      } else {
+        await Deno.remove(this.fileName)
+      }
+    } catch (e) { }
+  }
+}
+
+// Parse the output of `poetry` subcommands that support the `--dry-run` flag,
+// and submit the packages.
+//
+// Add the `-vvv` flags to get a detailed report of the dependency resolution
+// process. This way, we can track the actual lockfile changes, as specifying
+// `--dry-run` only would not output anything in combination with flags such
+// as `--lock` that do not perform the actual operations.
+async function poetryCheckDryRun(subcommand: string, args: string[]): boolean {
+  // Read and backup the current poetry lockfile contents.
+  const lockfileBackup = new FileBackup('poetry.lock')
+  const manifestBackup = new FileBackup('pyproject.toml')
+
+  await lockfileBackup.backup()
+  await manifestBackup.backup()
+
+  let process = Deno.run({
+    cmd: ['poetry', subcommand, '-vvv', '-n', '--dry-run', ...args.map(s => s.toString())],
+    stdout: 'piped',
+    stderr: 'piped',
+  })
+
+  await process.status()
+  await process.close()
+
+  // If it existed before, restore the previous contents of the lockfile;
+  // otherwise, delete it. This is a workaround to the fact that in poetry
+  // 1.1.x, the `--dry-run` argument does not prevent the lockfile from 
+  // being modified. This is not fixed as of poetry 1.1.14.
+  // Prudently, do the same for the manifest (pyproject.toml).
+  //
+  // See: https://github.com/python-poetry/poetry/pull/5718
+  await lockfileBackup.restoreOrDelete()
+  await manifestBackup.restoreOrDelete()
+
+  const output = new TextDecoder().decode(await process.output())
+  const packages = parseDryRun(output)
+
+  console.log(`Analyzing packages:`)
+  for (const { name, version } of packages) {
+    console.log(`  - ${name} ${version}`)
+  }
+  console.log()
```

Comment:
Does the user want to know this? I probably wouldn't mind but I'm not sure if this is a benefit or just annoying. Imagine this adds a billion packages, it would get extremely verbose.

Reply:
Changed this to print the first 10 items, and then a line that states `...and X more` in case the list exceeds 10 items. This way, the data is incomplete but not jarring to read, and the user can be made aware of the size of their dependency tree.

Reply:
Honestly I think this is the worst of both worlds. Now you still get your output spammed, but it's not even useful because there's no way to see **everything** that was installed.

I say either we go the `cargo` route and be very verbose, listing everything we got by default.

Or we stick to the ecosystem's own standards and hide the packages like `poetry` does.

I think I'd opt for the latter, to be consistent with the ecosystem. But I don't mind it much either way. I definitely don't like the half-way point with only showing a limited amount though.

Reply:
I will go for the latter.

Reply:
I also vote for displaying like poetry does. If the idea was that _something_ should be said then maybe a simple one-line output, at the correct spot, to say that Phylum is analyzing the dependencies would help to explain why the process is taking longer than when just using `poetry` directly.